google/cloud/iam/samples/iam_samples.cc

```diff
@@ -0,0 +1,627 @@
+  //! [END iam_create_role] [iam-create-role]
+  (argv.at(0), argv.at(1), {argv.begin() + 2, argv.end()});
+}
+
+void DeleteRole(std::vector<std::string> const& argv) {
+  if (argv.size() != 1 || argv.at(0) == "--help") {
+    throw google::cloud::testing_util::Usage("delete-role <role-name>");
+  }
+  //! [START iam_delete_role] [iam-delete-role]
+  namespace iam = google::cloud::iam;
+  [](std::string const& name) {
+    iam::IAMClient client(iam::MakeIAMConnection());
+    google::iam::admin::v1::DeleteRoleRequest request;
+    request.set_name(name);
+    auto response = client.DeleteRole(request);
+    if (!response) throw std::runtime_error(response.status().message());
+    std::cout << "Role successfully deleted: " << response->DebugString()
+              << "\n";
+  }
+  //! [END iam_delete_role] [iam-delete-role]
+  (argv.at(0));
+}
+
+void DisableServiceAccount(std::vector<std::string> const& argv) {
+  if (argv.size() != 1 || argv.at(0) == "--help") {
+    throw google::cloud::testing_util::Usage(
+        "disable-service-account <service-account-name>");
+  }
+  //! [START iam_disable_service_account] [iam-disable-service-account]
+  namespace iam = google::cloud::iam;
+  [](std::string const& name) {
+    iam::IAMClient client(iam::MakeIAMConnection());
+    google::iam::admin::v1::DisableServiceAccountRequest request;
+    request.set_name(name);
+    auto response = client.DisableServiceAccount(request);
+    if (!response.ok()) throw std::runtime_error(response.message());
+    std::cout << "ServiceAccount successfully disabled.\n";
+  }
+  //! [END iam_disable_service_account] [iam-disable-service-account]
+  (argv.at(0));
+}
+
+void EnableServiceAccount(std::vector<std::string> const& argv) {
+  if (argv.size() != 1 || argv.at(0) == "--help") {
+    throw google::cloud::testing_util::Usage(
+        "enable-service-account <service-account-name>");
+  }
+  //! [START iam_enable_service_account] [iam-enable-service-account]
+  namespace iam = google::cloud::iam;
+  [](std::string const& name) {
+    iam::IAMClient client(iam::MakeIAMConnection());
+    google::iam::admin::v1::EnableServiceAccountRequest request;
+    request.set_name(name);
+    auto response = client.EnableServiceAccount(request);
+    if (!response.ok()) throw std::runtime_error(response.message());
+    std::cout << "ServiceAccount successfully enabled.\n";
+  }
+  //! [END iam_enable_service_account] [iam-enable-service-account]
+  (argv.at(0));
+}
+
+void UpdateRole(std::vector<std::string> const& argv) {
+  if (argv.size() != 2) {
+    throw google::cloud::testing_util::Usage(
+        "update-role <role-name> <new-title>");
+  }
+  //! [START iam_edit_role] [iam-update-role]
+  namespace iam = google::cloud::iam;
+  [](std::string const& name, std::string const& title) {
+    iam::IAMClient client(iam::MakeIAMConnection());
+    google::iam::admin::v1::UpdateRoleRequest request;
+    request.set_name(name);
+    ::google::iam::admin::v1::Role role;
```

Comment:
We usually don't fully qualify these names w/ the leading `::`. Is that needed here? If not, removing it would make the code more consistent.